src/app/leverage/provider.tsx

```diff
@@ -406,99 +381,102 @@ export function LeverageProvider(props: { children: any }) {
+
+  const fetchSwapQuote = async () => {
+    if (!address) return null
+    if (!chainId || chainId === ARBITRUM.chainId) return null
+    if (!publicClient) return null
+    if (inputTokenAmount <= 0) return null
+    if (!indexToken) return null
+    const inputTokenPrice = await getTokenPrice(inputToken, chainId)
+    const outputTokenPrice = await getTokenPrice(outputToken, chainId)
+    return await getIndexQuote({
+      isMinting,
+      chainId,
+      address,
+      inputToken,
+      inputTokenAmount: inputValue,
+      inputTokenPrice,
+      outputToken,
+      outputTokenPrice,
+      nativeTokenPrice,
+      slippage: 0.1,
+    })
+  }
+
+  const { data: swapQuote, isFetching: isFetchingSwapQuote } = useQuery({
+    queryKey: [
+      'swap-quote',
+      {
         address,
-        inputToken,
-        inputTokenAmount: inputValue,
-        inputTokenPrice,
-        outputToken,
-        outputTokenPrice,
-        nativeTokenPrice,
-        slippage: 0.1,
-      })
-    }
-    const fetchQuote = async () => {
-      if (!address) return null
-      if (!chainId) return null
-      if (!provider || !publicClient) return null
-      if (inputTokenAmount <= 0) return null
-      if (!indexToken) return null
-      const inputTokenPrice = await getTokenPrice(inputToken, chainId)
-      const outputTokenPrice = await getTokenPrice(outputToken, chainId)
-      return await getFlashMintQuote({
-        isMinting,
-        account: address,
         chainId,
+        indexToken,
         inputToken,
-        inputTokenAmount: inputValue,
-        inputTokenAmountWei: inputTokenAmount,
-        inputTokenPrice,
-        outputToken,
-        outputTokenPrice,
-        slippage: 0.1,
-      })
-    }
+        inputTokenAmount: inputTokenAmount.toString(),
+        publicClient,
+      },
+    ],
+    queryFn: fetchSwapQuote,
+    enabled:
+      !!address &&
+      !!chainId &&
+      !!publicClient &&
+      !!indexToken &&
+      inputTokenAmount > 0,
+  })
 
-    const fetchQuotes = async () => {
-      setFetchingQuote(true)
-      try {
-        const [quoteResult, swapQuoteResult] = await Promise.allSettled([
-          fetchQuote(),
-          fetchSwapQuote(),
-        ])
-        setFlashmintQuote(
-          quoteResult.status === 'fulfilled' ? quoteResult.value : null,
-        )
-        setSwapQuote(
-          swapQuoteResult.status === 'fulfilled' ? swapQuoteResult.value : null,
-        )
-      } catch (error) {
-        console.error('Error fetching quotes', error)
-      } finally {
-        setFetchingQuote(false)
-      }
-    }
-    fetchQuotes()
-  }, [
-    address,
-    chainId,
-    indexToken,
-    inputToken,
-    inputTokenAmount,
-    inputValue,
-    isMinting,
-    nativeTokenPrice,
-    outputToken,
-    provider,
-    publicClient,
-    rpcUrl,
-  ])
+  const isFetchingQuote = useMemo(
+    () => isFetchingFlashMintQuote || isFetchingSwapQuote,
+    [isFetchingFlashMintQuote, isFetchingSwapQuote],
+  )
```

Comment:
nit: might be more performant to just use it 'raw' here, e.g.
`const isFetchingQuote = isFetchingFlashMintQuote || isFetchingSwapQuote `

related: https://kentcdodds.com/blog/usememo-and-usecallback